### ISSUE TITLE
TrackParticles is not available in Moving window 

### ISSUE THREAD

**Issue by XiangTangUSTC** [SmileiPIC/Smilei#569]
Dear Sirs
In 3D simulation, I want to track some particles for a long time, so I set the Moving window. But when the window starts moving, all parameters of the particle will become 0. Is there any error in my name list or post-processing.
The Moving window set：

MovingWindow(
    time_start = Lx,
    velocity_x = 0.9,
)

The DiagTrackParticles set：

DiagTrackParticles(
    species = "electron",
    every = 50,
    filter = my_filter,
    attributes = [
        "x", "y", "z",
        "px", "py", "pz",
        "weight",
    ],
)

In post-processing, I want to save the data in mat format. This is the post-processing code：

 mydata=array(sim.TrackParticles("electron", axes = ["x","y","z","px"],units=['um']).getData())
hdf5storage.savemat('pxdata.mat',{'data':mydata}, dormat='7.3', matlab_compatible = True) 

This may be a stupid and simple problem, but I can't solve it all the time. Look forward to your answers, thank you.

**Comment by mccoys** [comment]
When the window starts, all the data becomes 0 ? Or only for some particles?

Concerning the mat format, I don't understand the problem. Could you provide some details?

**Comment by XiangTangUSTC** [comment]
Sorry, I didn't describe it clearly .mat is a data format that can be read by MATLAB.
When the window starts to move, the Id of all particles will become 0. Then other parameters of particles (including x, y, z, px, py, pz, weight) will become undefined. For example, in the .mat file, the particle coordinates will become NaN. This may mean that the particle parameter information is lost.

**Comment by mccoys** [comment]
The particle coordinates become NaN when they are not in the box anymore. If they stay in the moving window, they should remain with their current coordinates.

You have a filter in your TrackParticles. This also means that the particles that do not pass the filter may disappear from the diagnostic (become NaN).

Concerning mat format, if this is not an issue in Smilei or happi, then it is not the right place to post the issue.

**Comment by XiangTangUSTC** [comment]
Hello, I have successfully solved this problem. 
After carefully reading the tutorial on moving windows on the website(https://smileipic.github.io/Smilei/Use/namelist.html), I found that moving windows is done by deleting a x_ min patch, then add a x_ max patch. Because of the particularity of my simulation, I only set one patch in the x direction to ensure that the load of each process is the same. When I set 64 patches in the x direction, the problem was solved. Thank you for your answer.
